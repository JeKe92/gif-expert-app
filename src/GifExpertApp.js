import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = []}) => {

    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <header>
                <h2 className="header-title">GifExpertApp</h2>
                <AddCategory setCategories={ setCategories }/>
                <hr /> 
            </header>
            <section>
                    {
                        categories.map( category => 
                        <GifGrid 
                            key={category}
                            category={category}
                        />)
                    }
            </section>
        </>
    )
}
