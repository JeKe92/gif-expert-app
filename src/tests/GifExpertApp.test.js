import React from 'react'
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe('Tests for GifExpertApp component', () => {
    

    test('should compare snapshot with component', () => {
            const categories = ['Love', 'God'];
            const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

            expect(wrapper).toMatchSnapshot();
            expect(wrapper.find('GifGrid').length).toBe( categories.length );
    });
    
});
