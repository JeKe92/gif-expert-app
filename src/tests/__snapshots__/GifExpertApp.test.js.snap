// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifExpertApp component should compare snapshot with component 1`] = `
<Fragment>
  <header>
    <h2
      className="header-title"
    >
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </header>
  <section>
    <GifGrid
      category="Love"
      key="Love"
    />
    <GifGrid
      category="God"
      key="God"
    />
  </section>
</Fragment>
`;
